### ISSUE TITLE
[webEXT] Add cookieStoreId option to browser.downloads APIs

### ISSUE THREAD

**Issue by Rumyra** [mdn/content#7747]
[Add cookieStoreId to browser.downloads](https://bugzilla.mozilla.org/show_bug.cgi?id=1669566)

## Acceptance Criteria

- [ ] The listed features are documented sufficiently on MDN
- [ ] BCD is updated
- [ ] Interactive example and data repos are updated if appropriate
- [ ] The content has been reviewed as needed

### For folks helping with Firefox related documentation

- [ ] Set bugs to `dev-doc-complete`
- [ ] Add entry to Firefox release notes if feature is enabled in release
or
- [ ] Add entry to Firefox experimental features page if feature is not yet enabled in release

## Features to document

- Add cookieStoreId option to browser.downloads APIs

## Related Gecko bugs

- [browser.downloads.download method only passes default container cookies](https://bugzilla.mozilla.org/show_bug.cgi?id=1669566).

**Comment by hamishwillee** [comment]
@caitmuenster @rebloor (G'day!) I've assigned to you, because this is webext and I found that you have already done most of this work in https://github.com/mdn/content/pull/8106 and  https://github.com/mdn/browser-compat-data/pull/12069

Very happy to hear it - I found the bug hard to parse ;-)

The remaining part from the release point of view is a release note. Are you guys planning on creating that?

**Comment by rebloor** [comment]
@hamishwillee Thanks for catching this one, I've added release notes details to https://github.com/mdn/content/pull/8106
